build/rust/BUILD.gn

```diff
@@ -44,6 +44,7 @@ cargo_build("rust_lib") {
     "//brave/components/challenge_bypass_ristretto:rust_lib",
     "//brave/components/filecoin:rust_lib",
     "//brave/components/json:rust_lib",
+    "//brave/components/nested_star:rust_lib",
```

Comment:
Do we have concrete plans to use nested star outside P3A? If not it probably makes more sense to make this part of the p3a component until we have clear requirements for a shared API. For example, what if I want non-nested STAR for a different application? A `nested_star` component doesn't help, yet it shouldn't itself use a separate chromium-level `star` component either.

Reply:
While we don't have concrete plans to use star outside of P3A, I thought it would be a good idea to have a component that abstracts the CXX binding from the rest of the P3A code. `adblock_rust_ffi` and `challenge_bypass_ristretto` are examples of this pattern. It seems that adding another component doesn't add much cost to the build process, so I'm not sure I see the harm in keeping a separate component.